# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1281046430 25200
#      Thu Aug 05 15:13:50 2010 -0700
# Node ID 09144fbb0e36ec8ab75003dc185a0b5b57b8ab21
# Parent  bf000a56a9857a3712e28fafc96682403bb9e1f8
Fix bug #30400 when bicubic called with small numbers of arguments.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2010-08-01 Marco Caliari <marco.caliari@univr.it>
+
+	* general/bicubic.m: Fix bug #30400 when bicubic called with small
+	numbers of arguments.
+
 2010-08-04  Jaroslav Hajek  <highegg@gmail.com>
 
 	* general/sortrows.m: Don't actually permute the rows if not
 	requested.
 
 2010-08-04  Ben Abbott <bpabbott@mac.com>
 
 	* /plot/print.m, plot/__print_parse_opts__.m, plot/__gnuplot_print__.m, 
diff --git a/scripts/general/bicubic.m b/scripts/general/bicubic.m
--- a/scripts/general/bicubic.m
+++ b/scripts/general/bicubic.m
@@ -172,17 +172,17 @@ function F = bicubic (X, Y, Z, XI, YI, e
   s = []; s2 = []; s3 = [];
 
   cs0 = cs0([1,1,1,1],:);
   cs1 = cs1([1,1,1,1],:);
   cs2 = cs2([1,1,1,1],:);
   cs3 = cs3([1,1,1,1],:);
 
   lent = length (ct0);
-  lens = length (cs0);
+  lens = columns (cs0);
   F = zeros (lent, lens);
   
   for i = 1:lent
     it = indt(i);
     int = [it, it+1, it+2, it+3];
     F(i,:) = ([ct0(i),ct1(i),ct2(i),ct3(i)]
               * (p(int,inds) .* cs0 + p(int,inds+1) .* cs1
                  + p(int,inds+2) .* cs2 + p(int,inds+3) .* cs3));
